### PR TITLE
docs: Windows vhd doc

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Adding a doc outlining motivations for and high level process outline of building Windows AKS VHDs

**Issue Fixed**:

n/a

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: docs/topics/windows-vhd.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Windows AKS Azure marketplace images
+
+## Motivation
+
+The primary motivation for producing AKS specific Windows VHDs is to improve the reliability and deployment time for the configuration phase of Windows nodes configured by aks-engine. 
+By performing expensive configuration options and pre-downloading installation artifacts as part of the VHD build process we can accomplish both of these goals.
+
+Producing and publishing AKS specific Windows VHDs to the Azure marketplace also allow us to get targeted windows patches to customers sooner. 
+Today the WIndows/Windows Server team only publish new images containing the latest B week patches. Several times recently C week patches contain fixes to issues customers are facing running Kubernetes on Windows nodes. This means customers would either need to wait an additional 3-4 weeks to get these fixes or another mechanism to developed these patches to Windows nodes would need to be developed.
```
Comment (PatrickLang) [inline docs/topics/windows-vhd.md]: it's probably worth calling out that if a C fix isn't deployed, the wait is for the next month's cumulative release (B) which includes the fix.

File: docs/topics/windows-vhd.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Windows AKS VHDs are produced using [packer](http://www.packer.io) in an [Azure DevOps](http://dev.azure.com) build pipeline.
+
+The build pipeline:
+- Runs a packer job which provisions a new VM in Azure based off a specified Windows Server image and:
+  - Installs/enables required Windows components/features
+  - Download multiple versions of components which users are allowed to specify the version of (Kubernetes, Docker, etc)
+  - Downloads commonly used container images
+  - Configure system settings (windows update, page file sizes, etc)
+- Runs E2E tests using the VHD produced in the same pipeline
```
Comment (PatrickLang) [inline docs/topics/windows-vhd.md]: AKS-Engine E2E

Note: A full E2E test pass of Kubernetes will come later

File: docs/topics/windows-vhd.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- Run the pipeline
+
+## Usage in aks-engine
+
+Both Windows and Linux nodes are configured by executing Custom Script Extensions as part of VM deployment operations.
+The scripts are generated by aks-engine which populates instance specific information from the [cluster definition API model](clusterdefinitions.md) and get embedded into the ARM templates produced by AKS engine. These scripts are enlightened to understand the work in conjunction with Windows AKS marketplace images to not duplicate configuration steps and/or utilize files cached during the VHD build process.
+
+[Parts/k8s/kuberneteswindowssetup.ps1](../../parts/k8s/kuberneteswindowssetup.ps1) and associated ps1 files are used as templates for the extension scripts.
+
+It is an explicit goal to maintain aks-engine compatibility with 'vanilla' Windows Server marketplace images (for most scenarios) and to treat the Windows AKS marketplace images as an optimization for the reasons stated above.
```
Comment (PatrickLang) [inline docs/topics/windows-vhd.md]: It's probably worth calling out that installing patches on the 'vanilla' images is a non-goal.

Comment (PatrickLang) [reply docs/topics/windows-vhd.md]: also rather than using the term 'vanilla' can you just add a reference to the actual marketplace image we're talking about? that should translate better :)

File: docs/topics/windows-vhd.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+The build pipeline:
+- Runs a packer job which provisions a new VM in Azure based off a specified Windows Server image and:
+  - Installs/enables required Windows components/features
+  - Download multiple versions of components which users are allowed to specify the version of (Kubernetes, Docker, etc)
+  - Downloads commonly used container images
+  - Configure system settings (windows update, page file sizes, etc)
+- Runs E2E tests using the VHD produced in the same pipeline
+- Optionally copies the VHD to another Azure storage account for extra validation and/or publishing 
+
```
Comment (PatrickLang) [inline docs/topics/windows-vhd.md]: also creates a readme file showing what's installed & cached

File: docs/topics/windows-vhd.md (hunk 1 above)
Comment (craiglpeters) [inline docs/topics/windows-vhd.md]: Nit: replace "WIndows" with "Windows"

File: docs/topics/windows-vhd.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+Producing and publishing AKS specific Windows VHDs to the Azure marketplace also allow us to get targeted windows patches to customers sooner. 
+Today the WIndows/Windows Server team only publish new images containing the latest B week patches. Several times recently C week patches contain fixes to issues customers are facing running Kubernetes on Windows nodes. This means customers would either need to wait an additional 3-4 weeks to get these fixes or another mechanism to developed these patches to Windows nodes would need to be developed.
+
+Lastly publishing AKS specific Windows VHDs allows us to perform adequate testing on new patches before allowing customers to customers to upgrade their Windows nodes.
+In the first half of 2019 monthly security fixes broke Kubernetes workloads on Windows a few different occasions forcing the the aks-engine team to start hard-coding the specific versions of the Windows Server marketplace images (instead of targeting latest). 
+This needs to be resolved by collaborating and testing with the Windows team before patches get released publicly regardless (see [Testing Private Fixes](#Testing-Private-Fixes) section below).
```
Comment (craiglpeters) [inline docs/topics/windows-vhd.md]: Maybe rephrase? How about something like, "Given the challenging nature of Windows fixes, we will work with the Windows team to include Kubernetes testing before public release of patches. In the mean time the VHD pipeline supports [Testing Private Fixes](#Testing-Private-Fixes)"
